storage/videx/ha_videx.h

```diff
@@ -0,0 +1,197 @@
+
+	void column_bitmaps_signal() override;
+
+	int open(const char *name, int mode, uint test_if_locked) override;
+
+	handler* clone(const char *name, MEM_ROOT *mem_root) override;
+
+	int close(void) override;
+
+	IO_AND_CPU_COST scan_time() override;
+	
+	IO_AND_CPU_COST rnd_pos_time(ha_rows rows) override;
+
+	int write_row(const uchar * buf) override;
+
+	int update_row(const uchar * old_data, const uchar * new_data) override;
+
+	int delete_row(const uchar * buf) override;
+
+	void unlock_row() override;
+
+	int index_next(uchar * buf) override;
+
+	int index_prev(uchar * buf) override;
+
+	int index_first(uchar * buf) override;
+
+	int index_last(uchar * buf) override;
+
+	int rnd_init(bool scan) override;
+
+	int rnd_end() override;
+
+	int rnd_next(uchar *buf) override;
+
+	int rnd_pos(uchar * buf, uchar *pos) override;
+
+	void position(const uchar *record) override;
+
+	int info(uint) override;
+
+	int extra(ha_extra_function operation) override;
+
+	int reset() override;
+
+	int external_lock(THD *thd, int lock_type) override;
+
+	THR_LOCK_DATA** store_lock(
+		THD*			thd,
+		THR_LOCK_DATA**		to,
+		thr_lock_type		lock_type) override;
+
+	ha_rows records_in_range(
+		uint                    inx,
+		const key_range*        min_key,
+		const key_range*        max_key,
+		page_range*             pages) override;
+
+	int create(
+		const char*		name,
+		TABLE*			form,
+		HA_CREATE_INFO*		create_info) override;
+
+	int delete_table(const char *name) override;
+
+	int rename_table(const char* from, const char* to) override;
+
+	int multi_range_read_init(
+		RANGE_SEQ_IF*		seq,
+		void*			seq_init_param,
+		uint			n_ranges,
+		uint			mode,
+		HANDLER_BUFFER*		buf) override;
+
+	int multi_range_read_next(range_id_t *range_info) override;
+
+	ha_rows multi_range_read_info_const(
+		uint			keyno,
+		RANGE_SEQ_IF*		seq,
+		void*			seq_init_param,
+		uint			n_ranges,
+		uint*			bufsz,
+		uint*			flags,
+		ha_rows                 limit,
+		Cost_estimate*		cost) override;
+
+	ha_rows multi_range_read_info(uint keyno, uint n_ranges, uint keys,
+		uint key_parts, uint* bufsz, uint* flags,
+		Cost_estimate* cost) override;
+
+	int multi_range_read_explain_info(uint mrr_mode, char *str, size_t size) override;
+
+	Item* idx_cond_push(uint keyno, Item* idx_cond) override;
+  
+	int info_low(uint flag, bool is_analyze);
+
+	/** The multi range read session object */
+	DsMrr_impl m_ds_mrr;
+	
+	/** Thread handle of the user currently using the handler;
+	this is set in external_lock function */
+	THD *m_user_thd;
+
+	/** Flags that specificy the handler instance (table) capability. */
+	Table_flags m_int_table_flags;
+
+	/** Index into the server's primary key meta-data table->key_info{} */
+	uint m_primary_key;
+
+	/** this is set to 1 when we are starting a table scan but have
+	not yet fetched any row, else false */
+	bool m_start_of_scan;
+	
+	/** If mysql has locked with external_lock() */
+	bool m_mysql_has_locked;
+};
+
+typedef float rec_per_key_t;
```

Comment:
Up to you, but you don't really have to have `ha_videx.h`, this can be moved to `ha_videx.cc` is you like.

Reply:
Sounds good. Haibo will submit a clean version shortly that merges videx.h into videx.cc. The separate videx.h file was just following the code style of example.h/example.cc from an early stage, so merging them is perfectly fine. Thanks Sergey.